src/imputation/tmi_imputation.py

```diff
@@ -0,0 +1,383 @@
+) -> pd.DataFrame:
+    """Calculate the mean and count for each target and imputation class combination
+    Returns a dictionary."""
+
+    # remove the "trim" tagged rows
+    trimmed_df = filter_by_column_content(df, f"{target_variable}_trim", ["dont_trim"])
+
+    # convert to floats for mean calculation
+    trimmed_df[f"{target_variable}"] = trimmed_df[f"{target_variable}"].astype("float")
+
+    dict_mean_growth_ratio = {}
+
+    # Add mean and count to dictionary
+    dict_mean_growth_ratio[f"{target_variable}_{unique_item}_mean"] = trimmed_df[f"{target_variable}"].mean()
+    dict_mean_growth_ratio[f"{target_variable}_{unique_item}_count"] = len(trimmed_df[f"{target_variable}"])
+
+    return dict_mean_growth_ratio
+
+
+def calculate_means(df, target_variable_list):
+    """Function to apply multiple steps to calculate the means for each target
+    variable.
+    Returns a dictionary of mean values and counts for each unique class and variable
+    Also returns a QA dataframe containing information on how trimming was applied"""
+    dfs_list = []
+
+    # Create an empty dict to store means
+    mean_dict = dict.fromkeys(target_variable_list)
+
+    copy_df = df.copy()
+
+    # Filter for clear statuses
+    clear_statuses = ["210", "211"]
+    filtered_df = filter_by_column_content(copy_df, "statusencoded", clear_statuses)
+
+    # Filter out imputation classes that are missing either "200" or "201"
+    filtered_df = filtered_df[~(filtered_df["imp_class"].str.contains("nan"))]
+    filtered_df = filtered_df[filtered_df["formtype"] == "0001"]
+
+    # Group by imp_class
+    grp = filtered_df.groupby("imp_class")
+    class_keys = list(grp.groups.keys())
+
+    for var in target_variable_list:
+        for k in class_keys:
+            # Get subgroup dataframe
+            subgrp = grp.get_group(k)
+            # Sort by target_variable, df['employees'], reference
+            sorted_df = sort(var, subgrp)
+
+            # Add trimming threshold marker
+            trimcheck_df = apply_trim_check(sorted_df, var)
+
+            # Apply trimming marker
+            trimmed_df = trim_bounds(trimcheck_df, var)
+
+            tr_df = trimmed_df.set_index("pre_index")
+
+            dfs_list.append(tr_df)
+            # Calculate mean and count # TODO: Rename this
+            means = calculate_mean(trimmed_df, k, var)
+
+            # Update full dict with values
+            if mean_dict[var] is None:
+                mean_dict[var] = means
+            else:
+                mean_dict[var].update(means)
+
+    df = pd.concat(dfs_list)
+    df["qa_index"] = df.index
+    df = df.groupby(["pre_index"], as_index=False).first()
+
+    return mean_dict, df
+
+
+def tmi_imputation(df, target_variables, mean_dict):
+    """Function to replace the not clear statuses with the mean value
+    for each imputation class"""
+    for var in target_variables:
+        df["imp_marker"] = "N/A"
```

Comment:
`df["imp_marker"] = "N/A"` does not need to be in the for loop. Once it's applied it would do this for the whole "imp_marker" column